servers/graphql-kotlin-ktor-server/src/main/kotlin/com/expediagroup/graphql/server/ktor/subscriptions/graphqlws/KtorGraphQLWebSocketProtocolHandler.kt

```diff
@@ -0,0 +1,208 @@
+                    else -> session.closeAsInvalidMessage("Unexpected operation ${message.type}")
+                }
+            }
+        } catch (ex: ClosedReceiveChannelException) {
+            logger.debug("Client disconnected")
+            subscriptionHooks.onDisconnect(session, context)
+        } catch (ex: Throwable) {
+            logger.error("Error on processing GraphQL subscription session", ex)
+            session.closeExceptionally(ex)
+        }
+    }
+
+    private suspend fun initializeSession(session: WebSocketServerSession): GraphQLContext? {
+        val initMessage = session.readMessageOrNull() ?: return null
+        if (initMessage.type != MessageTypes.GQL_CONNECTION_INIT) {
+            session.close(CloseReason(4401, "Unauthorized"))
+            return null
+        }
+        return try {
+            subscriptionHooks.onConnect(initMessage.payload, session)
+        } catch (ex: Throwable) {
+            logger.debug("Got error from onConnect hook, closing session", ex)
+            session.close(CloseReason(4403, "Forbidden"))
+            null
+        }
+    }
+
+    private suspend fun MutableMap<String, Job>.startSubscription(
+        message: SubscriptionOperationMessage,
+        session: WebSocketServerSession,
+        context: GraphQLContext
+    ) {
+        if (message.id == null) {
+            logger.debug("Missing id from subscription message")
+            session.closeAsInvalidMessage("Missing id from subscription message")
+            return
+        }
+        if (containsKey(message.id)) {
+            logger.debug("Already subscribed to operation {}", message.id)
+            session.close(CloseReason(4409, "Subscriber for ${message.id} already exists"))
+            return
+        }
+        if (message.payload == null) {
+            logger.debug("Missing payload from subscription message={}", message.id)
+            session.closeAsInvalidMessage("Missing payload from subscription message")
+            return
+        }
+
+        val request = try {
+            objectMapper.convertValue<GraphQLRequest>(message.payload)
+        } catch (ex: Throwable) {
+            logger.error("Error when parsing GraphQL request data", ex)
+            session.closeAsInvalidMessage("Error when parsing GraphQL request data")
+            return
+        }
+
+        try {
+            subscriptionHooks.onOperation(message.id, request, session, context)
+        } catch (ex: Throwable) {
+            logger.error("Error when running onOperation hook for operation={}", message.id, ex)
+            session.closeAsInvalidMessage(ex.message ?: "Error running onOperation hook for operation=${message.id}")
+            return
+        }
+
+        val subscriptionJob = session.launch {
+            subscriptionExecutor.executeSubscription(request, context)
+                .map {
+                    if (it.errors?.isNotEmpty() == true) {
```

Comment:
```suggestion
                    if (it.errors?.isNotEmpty()) {
```

Reply:
`it.errors?.isNotEmpty()` is `Boolean?`(nullable) 
that's why it compared to `true`

it's the same as `it.errors?.isNotEmpty() ?: false`